pkg/admission/mutator/shoot_test.go

```diff
@@ -0,0 +1,133 @@
+// limitations under the License.
+
+package mutator_test
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/gardener/gardener-extension-provider-aws/pkg/admission/mutator"
+	apisaws "github.com/gardener/gardener-extension-provider-aws/pkg/apis/aws"
+	"github.com/gardener/gardener-extension-provider-aws/pkg/aws"
+	"github.com/golang/mock/gomock"
+
+	extensionswebhook "github.com/gardener/gardener/extensions/pkg/webhook"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	mockclient "github.com/gardener/gardener/pkg/mock/controller-runtime/client"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+)
+
+var _ = Describe("Shoot mutator", func() {
+
+	Describe("#Mutate", func() {
+		const namespace = "garden-dev"
+
+		var (
+			shootMutator extensionswebhook.Mutator
+
+			ctrl  *gomock.Controller
+			c     *mockclient.MockClient
+			shoot *gardencorev1beta1.Shoot
+
+			ctx = context.TODO()
+
+			now     = metav1.Now()
+			gp2type = string(apisaws.VolumeTypeGP2)
+		)
+
+		BeforeEach(func() {
+			ctrl = gomock.NewController(GinkgoT())
+
+			shootMutator = mutator.NewShootMutator()
+
+			scheme := runtime.NewScheme()
+			Expect(gardencorev1beta1.AddToScheme(scheme)).To(Succeed())
+			Expect(shootMutator.(inject.Scheme).InjectScheme(scheme)).To(Succeed())
+
+			c = mockclient.NewMockClient(ctrl)
+			Expect(shootMutator.(inject.Client).InjectClient(c)).To(Succeed())
+			shoot = &gardencorev1beta1.Shoot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: namespace,
+				},
+				Spec: gardencorev1beta1.ShootSpec{
+					SeedName: pointer.String("aws"),
+					Provider: gardencorev1beta1.Provider{
+						Type: aws.Type,
+						Workers: []gardencorev1beta1.Worker{
+							{
+								Name: "worker-1",
+								Volume: &gardencorev1beta1.Volume{
+									VolumeSize: "50Gi",
+									Type:       pointer.String(gp2type),
+								},
+								Zones: []string{"zone1"},
+							},
+						},
+					},
+					Region: "us-west",
+					Networking: gardencorev1beta1.Networking{
+						Nodes: pointer.String("10.250.0.0/16"),
+						Type:  "calico",
+					},
+				},
+			}
+		})
+
+		Context("Mutate shoot", func() {
+			It("should return nil when shoot is in scheduled to new seed phase", func() {
+				shoot.Status.LastOperation = &gardencorev1beta1.LastOperation{
+					Description:    "test",
+					LastUpdateTime: metav1.Time{Time: metav1.Now().Add(time.Second * -1000)},
+					Progress:       0,
+					Type:           gardencorev1beta1.LastOperationTypeReconcile,
+					State:          gardencorev1beta1.LastOperationStateProcessing,
+				}
+				shoot.Status.SeedName = pointer.String("gcp")
+				err := shootMutator.Mutate(ctx, shoot, nil)
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
For testing Validators it is usually enough to check the error. But for Mutators, the mutator could actually return nil and still mutate the obj, right? Is it possible to add expectation that the given obj was not mutated? Otherwise we could have passing unit tests but mutations could still happen.
This comment applies also to the other tests.